whylog/config/__init__.py

```diff
@@ -13,6 +16,111 @@
 from whylog.config.rule import RegexRuleFactory
 
 
+class AbstractConfigFactory(object):
+    """
+    This factory is responsible for finding .whylog directory (whylog config directory) and basing on
+    found content directory creating object of subclass AbstractConfig. If not found then it creates minimal
+    .whylog version in current directory.
+    """
+    WHYLOG_DIR = '.whylog'
+    HOME_DIR = os.path.expanduser('~')
+    ETC_DIR = '/etc'
+    ASSISTANTS_DICT = {'regex': RegexAssistant}
+
+    @classmethod
+    @abstractmethod
+    def load_config(cls, path):
+        raise NotImplementedError
+
+    @classmethod
+    def _attach_whylog_dir(cls, path):
+        return os.path.join(path, cls.WHYLOG_DIR)
+
+    @classmethod
+    def _search_in_parents_directories(cls, path):
+        if os.path.isdir(cls.WHYLOG_DIR):
+            #current directory
+            return path
+        while True:
+            path, suffix = os.path.split(path)
+            if suffix == '':
+                return None
+            if os.path.isdir(cls._attach_whylog_dir(path)):
+                return path
+
+    @classmethod
+    def _check_concrete_directory(cls, path):
```

Comment:
This method is really `return os.path.isdir(path)`. Remove it and inline `os.path.isdir(path)`